### PR TITLE
Add CursorEditor tests and fix bugs

### LINKED ISSUE #1292: CursorEditor.EditValue throws InvalidCastException if `value` is not a Cursor
Should handle this gracefully as if it were a null value

```cs
public static IEnumerable<object[]> EditValue_TestData()
{
    yield return new object[] { null };
    yield return new object[] { "value" };
    yield return new object[] { Cursors.Default };
    yield return new object[] { new object() };
}

[Theory]
[MemberData(nameof(EditValue_TestData))]
public void CursorEditor_EditValue_ValidProvider_ReturnsValue(object value)
{
    var editor = new CursorEditor();
    var mockEditorService = new Mock<IWindowsFormsEditorService>(MockBehavior.Strict);
    var mockServiceProvider = new Mock<IServiceProvider>(MockBehavior.Strict);
    mockServiceProvider
        .Setup(p => p.GetService(typeof(IWindowsFormsEditorService)))
        .Returns(mockEditorService.Object)
        .Verifiable();
    mockEditorService
        .Setup(e => e.DropDownControl(It.IsAny<Control>()))
        .Verifiable();
    Assert.Equal(value, editor.EditValue(null, mockServiceProvider.Object, value));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Once());
    mockEditorService.Verify(e => e.DropDownControl(It.IsAny<Control>()), Times.Once());

    // Edit again.
    Assert.Equal(value, editor.EditValue(null, mockServiceProvider.Object, value));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Exactly(2));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Exactly(2));
}
```

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #1292
Contributes to #817

### CODE REVIEW

File: src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/CursorEditor.cs [hunk 1]
```diff
@@ -10,62 +10,54 @@
...
+        /// provider is provided so that any required editing services can be obtained.
         /// </summary>
         public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
         {
-            object returnValue = value;
-
-            if (provider == null)
-            {
-                return value;
-            }
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/CursorEditor.cs]: We should return fast, excessive nesting negatively affects readability and maintainability of the code. It also unnecessarily complicates testing.

File: src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/CursorEditor.cs [hunk 2]
```diff
@@ -10,62 +10,54 @@
...
-
-            IWindowsFormsEditorService edSvc = (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
-
-            if (edSvc != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/CursorEditor.cs]: Invert `if` to reduce nesting and revert the next line

**Comment by hughbe** [comment]
> Can you please clarify what the bug was?

If you called `new CursorEditor().EditValue(null, provider, new object())` then we'd throw an `InvalidCastException`

File: src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/CursorEditor.cs [hunk 3]
```diff
@@ -10,62 +10,56 @@
...
         /// </summary>
         public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
         {
-            object returnValue = value;
-
             if (provider == null)
             {
                 return value;
             }
+            if (!(provider.GetService(typeof(IWindowsFormsEditorService)) is IWindowsFormsEditorService edSvc))
```
Comment (hughbe) [inline src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/CursorEditor.cs]: Contribution to #817

**Comment by RussKie** [comment]
> > Can you please clarify what the bug was?
> 
> If you called `new CursorEditor().EditValue(null, provider, new object())` then we'd throw an `InvalidCastException`

Thank you. 
Next time please add information like this to a commit message and when raising a PR (there is a section for it)

**Pull request merged**